web/packages/teleterm/src/ui/AccessRequests/SelectorMenu.tsx

```diff
@@ -228,31 +241,26 @@ export function SelectorMenu() {
           <MenuItemSectionLabel
             css={`
               justify-content: space-between;
-              &:hover .refresh {
-                visibility: visible;
-              }
             `}
             color="text.muted"
             ml="8px"
             mr={2}
             my={1}
           >
             Available Requests
-            <ButtonText
-              size="small"
-              title="Refresh"
-              css={`
-                visibility: hidden;
-              `}
-              className="refresh"
-            >
-              <Icon.Refresh
-                size="small"
+            {fetchRequestsAttempt.status === 'processing' ? (
+              <Indicator delay="none" size="small" />
+            ) : (
+              <ButtonText
                 onClick={() => {
                   void fetchRequests();
```

Comment:
In the future, it'd be good to somehow modify `useDelayedRepeatedAttempt` so that it's possible to call the run function of the eager attempt while simultaneously bypassing the default behavior of `useDelayedRepeatedAttempt`.

I added a 2.5s delay to `fetchRequests` and it feels a bit off that you see any kind of feedback only after 400ms. Implicit calls to refresh the list (such as opening the menu) should depend on the delayed behavior, while explicit calls to refresh the list (like the button) should show feedback immediately.

Reply:
Ah I see what you mean, indeed it's a bit off. I left a TODO item, so we can come back to it.